bezier-rs/lib/src/lib.rs

```diff
@@ -70,18 +72,40 @@ impl Bezier {
 		}
 	}
 
-	/// Create a quadratic bezier curve that goes through 3 points
-	// #[inline]
-	pub fn quadratic_from_points(p1: DVec2, p2: DVec2, p3: DVec2, _t: f64) -> Self {
-		// TODO: Implement logic to get actual curve through the points
-		Bezier::from_quadratic_dvec2(p1, p2, p3)
+	/// Create a quadratic bezier curve that goes through 3 points, where the middle point will be at the corresponding position `t` on the curve.
+	/// Note that when `t = 0` or `t = 1`, the expectation is that the `point_on_curve` should be equal to `start` and `end` respectively.
+	/// In these cases, if the provided values are not equal, this function will use the `point_on_curve` as the `start`/`end` instead.
+	pub fn quadratic_through_points(start: DVec2, point_on_curve: DVec2, end: DVec2, t: f64) -> Self {
+		if t == 0. {
+			return Bezier::from_quadratic_dvec2(point_on_curve, point_on_curve, end);
+		} else if t == 1. {
+			return Bezier::from_quadratic_dvec2(start, point_on_curve, point_on_curve);
+		}
+		let [a, _, _] = utils::compute_abc_for_quadratic_through_points(start, point_on_curve, end, t);
+		Bezier::from_quadratic_dvec2(start, a, end)
 	}
 
-	/// Create a cubic bezier curve that goes through 3 points. d1 represents the strut.
-	// #[inline]
-	pub fn cubic_from_points(p1: DVec2, p2: DVec2, p3: DVec2, _t: f64, _d1: f64) -> Self {
-		// TODO: Implement logic to get actual curve through the points
-		Bezier::from_quadratic_dvec2(p1, p2, p3)
+	/// Create a cubic bezier curve that goes through 3 points, where the middle point will be at the corresponding position `t` on the curve.
+	/// Note that when `t = 0` or `t = 1`, the expectation is that the `point_on_curve` should be equal to `start` and `end` respectively.
+	/// In these cases, if the provided values are not equal, this function will use the `point_on_curve` as the `start`/`end` instead.
+	/// * `strut` is a representation of the how wide the resulting curve will be by designating the distance between the `e1` and `e2` defined in [the projection identity section](https://pomax.github.io/bezierinfo/#abc) of Pomax's bezier curve primer.
```

Comment:
The asterisk represent a bullet list item. I chose to format the rustdoc message in this way because I wanted the explanation of the function parameter to be separate from the core function description. When it's yet another sentence in the paragraph, it's easier to miss. But there might be some better convention that I'm not aware of, so please let me know if there's a different format that you prefer.

![image](https://user-images.githubusercontent.com/30204508/176544074-fbc6056d-3646-4cc9-b4e6-3ec81d120514.png)

Reply:
Ok, if you think it helps readability, that's good. But please use a dash instead of an asterisk since that's our preferred markdown format.